db_stress_tool/no_batched_ops_stress.cc

```diff
@@ -606,6 +612,9 @@ class NonBatchedOpsStressTest : public StressTest {
         // into the transaction. This will create an overlap with the MultiGet
         // keys and exercise some corner cases in the code
         if (thread->rand.OneIn(10)) {
+          ryw_check_locks.emplace_back(
+              new MutexLock(thread->shared->GetMutexForKey(
+                  rand_column_families[0], rand_keys[i])));
```

Comment:
Do we need locking? It looks like we remembered the exact values we wrote to an uncommitted transaction (`read_your_write_values`). I thought such writes would always take precedence when reading from that transaction.

Reply:
Thank you for the suggestion. I added the locks when occasionally there are verification failure, I thought it's race condition. After some investigation, I think the root cause is previously the transaction could be overwriting keys that do not allow overwrite, in that case, a misused`SingleDelete` in transaction rollback can cause these inconsistency. I have remove the key locks now.